packages/js/product-editor/src/blocks/variation-items/edit.tsx

```diff
@@ -19,11 +35,160 @@ export function Edit( {
 		isInSelectedTab?: boolean;
 	};
 } ) {
+	const noticeDimissed = useRef( false );
+	const { invalidateResolution } = useDispatch(
+		EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
+	);
+	const productId = useEntityId( 'postType', 'product' );
 	const blockProps = useBlockProps();
+	const [ productStatus ] = useEntityProp< string >(
+		'postType',
+		'product',
+		'status'
+	);
+
+	const totalCountWithoutPriceRequestParams = useMemo(
+		() => ( {
+			product_id: productId,
+			order: 'asc',
+			orderby: 'menu_order',
+			has_price: false,
+		} ),
+		[ productId ]
+	);
+
+	const { totalCountWithoutPrice } = useSelect(
+		( select ) => {
+			const { getProductVariationsTotalCount } = select(
+				EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
+			);
+
+			return {
+				totalCountWithoutPrice:
+					getProductVariationsTotalCount< number >(
+						totalCountWithoutPriceRequestParams
+					),
+			};
+		},
+		[ productId ]
+	);
+
+	const {
+		updateUserPreferences,
+		variable_items_without_price_notice_dismissed:
+			itemsWithoutPriceNoticeDismissed,
+	} = useUserPreferences();
+
+	const { ref: variationTableRef } = useValidation< Product >(
```

Comment:
How does this work? I see it's only returning a ref, which you're passing to VariationsTable. When is the regularPriceValidator function called?

Reply:
The regular validator is called when you click either `saveDraft` or publish/add.
It is also possible to trigger it manually, but I guess that is not necessary in our case.

The `ref` is to technically add a focus to the error field. This currently means it focuses the variation table, which is probably something we want to change in the future. I could also remove the use of `ref` if you wanted?